src/main/java/com/cloudbees/jenkins/support/api/BaseCommandOutputContent.java

```diff
@@ -0,0 +1,91 @@
+package com.cloudbees.jenkins.support.api;
+
+import com.cloudbees.jenkins.support.AsyncResultCache;
+import hudson.Functions;
+import hudson.model.Node;
+import hudson.remoting.VirtualChannel;
+import jenkins.model.Jenkins;
+import jenkins.security.MasterToSlaveCallable;
+import org.apache.commons.io.IOUtils;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.WeakHashMap;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+import java.util.logging.Logger;
+
+/**
+ * Common logic to CommandOutputContent and UnFilteredCommandOutputContent.
+ */
+@Restricted(NoExternalUse.class)
+class BaseCommandOutputContent {
+    private static final Logger LOGGER = Logger.getLogger(BaseCommandOutputContent.class.getName());
+
+    // It was previously in CommandOutputContent, but the constructor is private and there isn't any method returning
+    // an object of this class. As no external use is expected, it's been moved here.
+    private static class CommandLauncher extends MasterToSlaveCallable<String, RuntimeException> {
+        final String[] command;
+
+        private CommandLauncher(String... command) {
+            this.command = command;
+        }
+
+        public String call() {
+            StringWriter bos = new StringWriter();
```

Comment:
Closing?

Reply:
It has no effect, it's a string buffer, not a real stream. Anyway, the code was there, I just moved it to a base class.